src/provider/Signal.ts

```diff
@@ -0,0 +1,262 @@
+// tslint:disable-next-line:no-any
+type Context = any;
+
+/**
+ * A 'library' of common aggregators.
+ *
+ * Pass one of these functions to a `Signal[n]` constructor to add the described behaviour to that signal.
+ */
+export class Aggregators {
+    /**
+     * Basic "pass-through" aggregator. Will pass an array of values back to the signal emitter, one per callback.
+     *
+     * Note that this means the size of the returned array depends on the number of listeners attached.
+     */
+    public static ARRAY<T>(items: T[]): T[] {
+        return items;
+    }
+
+    /**
+     * If a signal allows listeners to return a promise, this aggregator will await all the received promises and then
+     * return an array of the resolved values.
+     *
+     * This aggregator is intended for use with non-void promises, where the value returned by the promise is
+     * meaningful. If the return value of the promise is not significant, consider {@link AwaitVoid}.
+     */
+    public static AWAIT<T>(items: Promise<T>[]): Promise<T[]> {
+        return Promise.all(items);
+    }
+
+    /**
+     * An aggregator for use with Promise<void> signals. Will await all promises that are returned by callback
+     * functions, but discard the (void) return values.
+     */
+    public static async AWAIT_VOID(items: Promise<void>[]): Promise<void> {
+        await Promise.all(items);
+    }
+}
+
+/**
+ * Signature for an aggregator function.
+ *
+ * On signals that allow callbacks to return a value (i.e. where `R != void`), an aggregator can be used to determine
+ * how the values returned by the signal callbacks are returned back to the code that emitted the signal.
+ *
+ * If no aggregator is provided, the `emit()` function on a signal has a `void` return type.
+ */
+export type Aggregator<R, R2> = (items: R[]) => R2;
+
```

Comment:
I get what this is doing, but the name is a little misleading, given none of these actually aggregate anything; i.e. there is nothing of the signature T[] => T. I think a better name is sthing like 'PostProcessor'. Thoughts?